### PR TITLE
Set new default approval mode based on repo visibility

### PR DESCRIPTION
**Pull request by 6543**
if not already altered by user

build upon  #4429  merged and released with v2.8.0

### CODE REVIEW

File: server/api/repo.go [hunk 1]
```diff
@@ -259,12 +260,9 @@ func PatchRepo(c *gin.Context) {
...
 			return
 		}
-	} else if in.IsGated != nil { // TODO: remove isGated in next major release
-		if *in.IsGated {
-			repo.RequireApproval = model.RequireApprovalAllEvents
-		} else {
-			repo.RequireApproval = model.RequireApprovalForks
-		}
+	} else if in.IsGated != nil {
+		c.String(http.StatusBadRequest, "'gated' option was used, use 'require_approval'")
```
Comment (6543) [inline server/api/repo.go:264]: suggestions for better text aprechiated

File: server/store/datastore/migration/020_set-new-defaults-for-require_approval.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"src.techknowlogick.com/xormigrate"
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+var setNewDefaultsForRequireApproval = xormigrate.Migration{
+	ID: "set-new-defaults-for-require-approval",
+	MigrateSession: func(sess *xorm.Session) (err error) {
+		const (
+			RequireApprovalNotSet    string = ""
```
Comment (6543) [inline server/store/datastore/migration/020_set-new-defaults-for-require_approval.go:29]: RequireApprovalOldNotGated = old_not_gated

File: server/store/datastore/migration/019_gated_to_require_approval.go [hunk 3]
```diff
@@ -26,10 +26,8 @@ var gatedToRequireApproval = xormigrate.Migration{
 	ID: "gated-to-require-approval",
 	MigrateSession: func(sess *xorm.Session) (err error) {
 		const (
-			RequireApprovalNone         string = "none"
-			RequireApprovalForks        string = "forks"
-			RequireApprovalPullRequests string = "pull_requests"
-			RequireApprovalAllEvents    string = "all_events"
+			RequireApprovalNotSet    string = ""
```
Comment (6543) [inline server/store/datastore/migration/019_gated_to_require_approval.go:29]: RequireApprovalOldNotGated = old_not_gated

**Comment by 6543** [comment]
@anbraten your backport was merged ... so this should be merged soon too ...

File: cli/repo/repo_update.go [hunk 4]
```diff
@@ -92,29 +91,18 @@ func repoUpdate(ctx context.Context, c *cli.Command) error {
...
-	// TODO: remove isGated in next major release
+
+	// TODO: remove in next release
 	if c.IsSet("gated") {
-		if gated {
-			patch.RequireApproval = &woodpecker.RequireApprovalAllEvents
-		} else {
-			patch.RequireApproval = &woodpecker.RequireApprovalNone
-		}
+		return fmt.Errorf("'gated' option was used, use 'require-approval'")
```
Comment (pat-s) [inline cli/repo/repo_update.go:97]:
```suggestion
		return fmt.Errorf("'gated' option has been set in version 2.8, use 'require-approval' in >= 3.0")
```

File: docs/src/pages/migrations.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ This will be the next version of Woodpecker.
 
 ## User migrations
 
+- `gated` option was replaced by `require-approval`
```
Comment (pat-s) [inline docs/src/pages/migrations.md:13]:
```suggestion
- `gated` has been replaced by `require-approval`
```

File: server/api/repo.go [hunk 6]
```diff
@@ -258,12 +259,9 @@ func PatchRepo(c *gin.Context) {
...
```
Comment (pat-s) [inline server/api/repo.go:263]:
```suggestion
		c.String(http.StatusBadRequest, "'gated' option has been set in version 2.8, use 'require-approval' in >= 3.0")
```

**Comment by pat-s** [comment]
@anbraten Waiting for you (?)

File: server/store/datastore/migration/019_gated_to_require_approval.go [hunk 7]
```diff
@@ -26,10 +26,8 @@ var gatedToRequireApproval = xormigrate.Migration{
 	ID: "gated-to-require-approval",
```
Comment (anbraten) [inline server/store/datastore/migration/019_gated_to_require_approval.go:26]: Would suggest to copy the exact migration from 2.8 here. It is currently missing the removal from the `gated` column. https://github.com/woodpecker-ci/woodpecker/blob/985a5ae73bf31e93e58f4c123a974b1de5bce556/server/store/datastore/migration/034_gated_to_require_approval.go

**Review by anbraten — COMMENTED** [review]
The migration still seems to miss the deletion of the gated column. Will do a full review later today.

**Comment by 6543** [comment]
> The migration still seems to miss the deletion of the gated column. Will do a full review later today.

this now is a 1:1 copy of what is in v2.8

**Comment by 6543** [comment]
> > The migration still seems to miss the deletion of the gated column. Will do a full review later today.
> 
> this now is a 1:1 copy of what is in v2.8

nop look at `server/store/datastore/migration/019_gated_to_require_approval.go` there it is

**Comment by anbraten** [comment]
> nop look at server/store/datastore/migration/019_gated_to_require_approval.go there it is

Ahh perfect. It was now hidden as we had it in next already.

**Review by anbraten — APPROVED** [review]

**Pull request merged**